include/utils/types/string_switch.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher
+
+#ifndef UTILS_TYPES_STRING_SWITCH_H
+#define UTILS_TYPES_STRING_SWITCH_H
+
+#include <string_view>
+
+namespace cura::utils
+{
+
+// Source: https://learnmoderncpp.com/2020/06/01/strings-as-switch-case-labels/
+inline constexpr uint64_t hash_djb2a(const std::string_view value)
+{
+    uint64_t hash{ 5381 };
+    for (unsigned char c : value)
+    {
+        hash = ((hash << 5) + hash) ^ c;
+    }
+    return hash;
+}
+
+inline constexpr uint64_t hash_enum(const std::string_view value)
+{
+    constexpr uint64_t plugin_namespace_sep_location{ 6 };
+    if (value.size() > plugin_namespace_sep_location && value.at(plugin_namespace_sep_location) == ':')
+    {
+        return hash_djb2a("plugin");
+    }
+    return hash_djb2a(value);
+}
+
+constexpr inline auto operator""_sw(const char* str, size_t len)
```

Comment:
In the above functions, you use `inline constexpr`, here you use `constexpr inline`. Pick a side, will ya? ;-)

Reply:
I will get my changes `inline`